cmd/kube-controller-manager/app/certificates.go

```diff
@@ -71,6 +76,9 @@ func startCSRSigningController(ctx ControllerContext) (http.Handler, bool, error
 	keyUsesDefault = (ctx.ComponentConfig.CSRSigningController.ClusterSigningKeyFile == kubeoptions.DefaultClusterSigningKeyFile)
 
 	switch {
+	case anySpecificFilesSet(ctx.ComponentConfig.CSRSigningController):
+		// in this case we continue because the default keys will not be used. Their status doesn't matter anymore.
```

Comment:
does this reflect what we discussed last week about disallowing combined use of the general ca/key files and specific signer loop ca/key files?

Reply:
> does this reflect what we discussed last week about disallowing combined use of the general ca/key files and specific signer loop ca/key files?

Yes it does.

Reply:
> does this reflect what we discussed last week about disallowing combined use of the general ca/key files and specific signer loop ca/key files?

correct.  Specifying a cert/key pair for any specific loop will prevent the general (old flag) from being used.

Reply:
would this be a good time to remove the default values for the general flags (as we've been indicating we will in the warning below) and error here if they set both the general and specific flags?

Reply:
(would remove cert/key file defaults in RecommendedDefaultCSRSigningControllerConfiguration and this entire switch statement)